### PR TITLE
Updates to allow tests to pass

### PR DESCRIPTION
**Pull request by s-good**
I've made various changes that fix inconsistencies with the original ECMWF code. The tests now all pass.

The new code is slower than the old code. I've removed all the file output I could find but there is still something slowing it down. These are the test output times from the old code:

>    Start 1: atlas_test_orca_plugin
> 1/7 Test 1: atlas_test_orca_plugin ............   Passed    0.33 sec
>    Start 2: atlas_test_orca_grid_specs
>2/7 Test 2: atlas_test_orca_grid_specs ........   Passed    0.08 sec
>    Start 3: atlas_test_orca_grid
>3/7 Test 3: atlas_test_orca_grid ..............   Passed    0.16 sec
>    Start 4: atlas_test_orca_mesh
>4/7 Test 4: atlas_test_orca_mesh ..............   Passed    1.24 sec
>    Start 5: atlas_test_orca_valid_elements
>5/7 Test 5: atlas_test_orca_valid_elements ....   Passed    4.09 sec
>    Start 6: atlas_test_orca_polygon_locator
>6/7 Test 6: atlas_test_orca_polygon_locator ...   Passed   11.69 sec
>    Start 7: atlas_test_orca_mesh_boundaries
>7/7 Test 7: atlas_test_orca_mesh_boundaries ...   Passed   48.54 sec

And here is the new code times:

> Start  1: atlas_test_orca_plugin
> 1/13 Test  1: atlas_test_orca_plugin ........................   Passed    0.38 sec
>      Start  2: atlas_test_orca_grid_specs
> 2/13 Test  2: atlas_test_orca_grid_specs ....................   Passed    0.10 sec
>      Start  3: atlas_test_orca_grid
> 3/13 Test  3: atlas_test_orca_grid ..........................   Passed    2.46 sec
>      Start  4: atlas_test_orca_mesh
> 4/13 Test  4: atlas_test_orca_mesh ..........................   Passed    2.35 sec
>      Start  5: atlas_test_orca_valid_elements
> 5/13 Test  5: atlas_test_orca_valid_elements ................   Passed    9.05 sec
>      Start  6: atlas_test_orca_polygon_locator
> 6/13 Test  6: atlas_test_orca_polygon_locator ...............   Passed   18.33 sec
>      Start  7: atlas_test_orca_mesh_boundaries_NOMPI
> 7/13 Test  7: atlas_test_orca_mesh_boundaries_NOMPI .........   Passed   74.10 sec
>      Start  8: atlas_test_orca_mesh_boundaries_MPI2
> 8/13 Test  8: atlas_test_orca_mesh_boundaries_MPI2 ..........   Passed   67.94 sec
>      Start  9: atlas_test_orca_surrounding_rectangle_NOMPI
> 9/13 Test  9: atlas_test_orca_surrounding_rectangle_NOMPI ...   Passed    0.38 sec
>      Start 10: atlas_test_orca_surrounding_rectangle_MPI2
>10/13 Test 10: atlas_test_orca_surrounding_rectangle_MPI2 ....   Passed    1.97 sec
>      Start 11: atlas_test_orca_surrounding_rectangle_MPI4
>11/13 Test 11: atlas_test_orca_surrounding_rectangle_MPI4 ....   Passed    0.92 sec
>      Start 12: atlas_test_orca_local_grid_NOMPI
>12/13 Test 12: atlas_test_orca_local_grid_NOMPI ..............   Passed    0.10 sec
>      Start 13: atlas_test_orca_local_grid_MPI2
>13/13 Test 13: atlas_test_orca_local_grid_MPI2 ...............   Passed    0.57 sec

### CODE REVIEW

File: src/tests/test_orca_mesh_boundaries.cc [hunk 1]
```diff
@@ -41,17 +41,17 @@ namespace test {
...
+        "eORCA025_T",  //
+        "eORCA12_T",  //
     };
     auto distributionNames = std::vector<std::string>{
       "serial",
       "checkerboard",
-      // "equal_regions",
-      // "equal_area",
+      "equal_regions",
+      // "equal_area", // This is not a valid option. Should it be equal_bands?
```
Comment (twsearle) [inline src/tests/test_orca_mesh_boundaries.cc]: Are you using the head of atlas develop for your build? I am pretty sure equal_area was added fairly recently. I am not sure but I think equal_bands wouldn't work for us.

**Comment by twsearle** [comment]
when you say this is slower, are you testing vs the head of develop? The base `atlas_test_orca_mesh_boundaries` uses 2 processors, so should match in timings with our `MPI2` version. Having said that I would not at all be surprised if this is slower as I intentionally chose to ignore performance in the interests of improving my understanding. If we can get halos working and prove that we have resolved the issues with correctness in sith, we can revisit performance again and make improvements. Certainly for our use case the meshgeneration step hasn't been a huge performance issue, and seems to scale with MPI fairly well. Whereas I/O and interpolation are where we lose a lot of our time.

File: src/tests/test_orca_mesh_boundaries.cc (hunk 1 above)
Comment (s-good) [reply src/tests/test_orca_mesh_boundaries.cc]: That explains it, thanks. I'll remove my comment.

**Comment by s-good** [comment]
> when you say this is slower, are you testing vs the head of develop? The base `atlas_test_orca_mesh_boundaries` uses 2 processors, so should match in timings with our `MPI2` version. Having said that I would not at all be surprised if this is slower as I intentionally chose to ignore performance in the interests of improving my understanding. If we can get halos working and prove that we have resolved the issues with correctness in sith, we can revisit performance again and make improvements. Certainly for our use case the meshgeneration step hasn't been a huge performance issue, and seems to scale with MPI fairly well. Whereas I/O and interpolation are where we lose a lot of our time.

Yes, my timing were versus develop. I agree that the difference isn't enough to be a problem. I was more worried that something was going wrong, but it sounds like it is not completely unexpected.

**Pull request merged**